add.go

```diff
@@ -100,6 +106,18 @@ func (b *Builder) Add(destination string, extract bool, source ...string) error
 	if len(source) > 1 && (destfi == nil || !destfi.IsDir()) {
 		return errors.Errorf("destination %q is not a directory", dest)
 	}
+	// Find out which user/group should the destination belong to.
```

Comment:
Create a separate function 
```
func (b *Builder) user(options AddAndCopyOptions) (string,  error) {
	// If this information is passed in through --chown, use it;
	// otherwise use the user/group as whom the container should run.
	if options.Chown != "" {
		return getUser(mountPoint, options.Chown)
	}
	 return getUser(mountPoint, b.User())
}
```

Reply:
That's right! Thanks!